### DISCUSSION TITLE
Debugging Reverse Engineered Code in Java

### DISCUSSION THREAD

**Discussion by Gcode-git** [community/community#130560] [Programming Help] [answered]
### Body

I've been working on reverse engineering a Java application, and I've encountered a bug that I can't seem to fix. The issue is with a method that calculates the factorial of a number. Here's the original code I reverse engineered

`public int calculateFactorial(int n) {
    if (n == 0) {
        return 1;
    }
    return n * calculateFactorial(n - 1);
}
`

When I input large numbers, it throws a StackOverflowError. Any suggestions on how to fix this?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by johnraivenolazo** [comment, marked as answer]
The issue you're experiencing is due to the recursion depth when you input large numbers, which causes a StackOverflowError. To handle large numbers more efficiently, you can use an iterative approach instead of recursion.

**Reply by Gcode-git** [reply]
Thanks! I tried looping through it and its now working

**Reply by ebndev** [reply]
Hi @JohnRaivenOlazo, thanks for taking the time to answer!

@Gcode-git glad you were able to resolve your issue 😄

**Comment by asdacosta** [comment]
Rewrite the method using a loop to avoid recursion. This is a more efficient approach for large factorials.

```
public int calculateFactorial(int n) {
    if (n == 0) {
        return 1;
    }
    int result = 1;
    for (int i = 1; i <= n; i++) {
        result *= i;
    }
    return result;
}
```

**Reply by Gcode-git** [reply]
Its working now man, thankssss
